### PR TITLE
Replace deprecated calls to `time.clock()` by `time.perf_counter()`

### LINKED ISSUE #2163: time.clock() is deprecated
Should remove and replace in timings module.

### PR DESCRIPTION
**Pull request by StreakyCobra**
Close #2163

### CODE REVIEW

File: xonsh/timings.py [hunk 1]
```diff
@@ -46,10 +46,10 @@ def clocku():
             """clocku() -> floating point number
             Return the *USER* CPU time in seconds since the start of the process.
             This is done via a call to resource.getrusage, so it avoids the
-            wraparound problems in time.clock()."""
+            wraparound problems in time.perf_counter()."""
```
Comment (StreakyCobra) [inline xonsh/timings.py]: I'm not sure to understand what are the «wraparound problems» there, so I don't know if a deeper refactoring should be done, or if the comment still make sense after changing the call to `time.perf_counter()`.

**Comment by melund** [comment]
Thanks for fixing this @StreakyCobra .  You probably need to remove the ` if ON_WINDOWS else 0.0` part here: 

https://github.com/xonsh/xonsh/blob/master/xonsh/timings.py#L231

This is used for the `xonsh --timings` command. 

As I understand the new `perf_counter` doesn't have a well-defined starting time. Previously, that was only an issue on Windows, while `clock()` on Linux would always be zero when the process started.

File: xonsh/timings.py (hunk 1 above)
Comment (melund) [inline xonsh/timings.py]: I don't know what it refers to either. Probably not an issue, but @scopatz may know.

File: news/fix_time_clock.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+**Changed:** None
+
+**Deprecated:** None
+
+**Removed:** None
+
+**Fixed:**
+
+* Replace deprecated calls to `time.clock()` by calls to `time.perf_counter()`.
```
Comment (melund) [inline news/fix_time_clock.rst]: This has to be ` ``time.clock()`` ` in restructured text.

Comment (StreakyCobra) [inline news/fix_time_clock.rst]: Indeed!

**Comment by StreakyCobra** [comment]
@melund:

> You probably need to remove the if ON_WINDOWS else 0.0 part

Not sure to understand all this, I'm not experienced with xonsh. For me it should already have been `clock()` on linux instead of setting it manually to `0.0`, event before this change. I've put this in a separate if you don't want to include it.

**Comment by scopatz** [comment]
> Not sure to understand all this, I'm not experienced with xonsh. For me it should already have been clock() on linux instead of setting it manually to 0.0 

This was just a small performance boost since clock() was guaranteed to be zero

File: news/fix_time_clock.rst [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+**Deprecated:** None
+
+**Removed:** None
+
+**Fixed:**
+
+* Replace deprecated calls to ``time.clock()`` by calls to
+  ``time.perf_counter()``.
+* Use ``clock()`` to set the start time of ``_timings`` in non-windows instead
+  of manually setting it to `0`.
```
Comment (scopatz) [inline news/fix_time_clock.rst]: Still have a single backtick around the zero :)

Comment (StreakyCobra) [inline news/fix_time_clock.rst]: Rhaaaaaaaaa

**Comment by StreakyCobra** [comment]
> This was just a small performance boost since clock() was guaranteed to be zero

@scopatz So this mean it should be kept then, no?

**Comment by scopatz** [comment]
@StreakyCobra What you have in the PR right now is correct because of the differences between clock() and `perf_counter()`  clock() was guaranteed to start at zero.

**Comment by StreakyCobra** [comment]
Ok got it :+1:

File: xonsh/timings.py (hunk 1 above)
Comment (scopatz) [inline xonsh/timings.py]: I think this is fine to keep as it is. The wraparound comment is not valid for `perf_counter()`, but still makes sense for clock().  You should either restore the comment to say clock() (which we are still avoiding) or remove the whole comment.  

The problem with clock is that it uses uint32s. So it was possible to reset times back to zero if you got to big.  `perf_counter()` uses floats under the covers and so doesn't have this issue.

**Comment by scopatz** [comment]
Thanks a million for putting this in! I really appreciate it!

File: xonsh/timings.py (hunk 1 above)
Comment (StreakyCobra) [inline xonsh/timings.py]: So, as `clock` is deprecated and won't be used again anyway, we can probably just remove the comment?

Comment (scopatz) [inline xonsh/timings.py]: probably

File: news/fix_time_clock.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+**Removed:** None
+
+**Fixed:**
+
+* Replace deprecated calls to ``time.clock()`` by calls to
+  ``time.perf_counter()``.
+* Use ``clock()`` to set the start time of ``_timings`` in non-windows instead
+  of manually setting it to ``0.0``.
+
+**Security):** None
```
Comment (scopatz) [inline news/fix_time_clock.rst]: Please remove the extra parentheses here.

Comment (StreakyCobra) [inline news/fix_time_clock.rst]: What the hell… done :+1:

**Comment by melund** [comment]
This looks good @StreakyCobra. Thanks a lot for fixing this.

**Pull request merged**